### ISSUE TITLE
tryReuseStructureFromOldProgram() doesn't detect options.typeRoots changes correctly

### ISSUE THREAD

**Issue by jeffreymorlan** [Microsoft/TypeScript#11805]
In src/compiler/program.ts `tryReuseStructureFromOldProgram()`:
```ts
!arrayIsEqualTo(oldOptions.typeRoots, oldOptions.typeRoots) ||
```
This check is ineffective because it's comparing the old array to itself.

**Comment by RyanCavanaugh** [comment]
derp

**Comment by RyanCavanaugh** [comment]
😞
